hazelcast/src/main/java/com/hazelcast/internal/util/phonehome/MapInfoCollector.java

```diff
@@ -55,52 +60,115 @@
     }
 
     private long countMapUsingEviction() {
-        return mapConfigs.stream().filter(mapConfig -> mapConfig != null
-                && mapConfig.getEvictionConfig().getEvictionPolicy() != EvictionPolicy.NONE).count();
+        return mapConfigs.stream().map(MapConfig::getEvictionConfig)
+                .filter(evictionConfig -> evictionConfig.getEvictionPolicy() != EvictionPolicy.NONE).count();
     }
 
     private long countMapWithNativeInMemoryFormat() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && mapConfig.getInMemoryFormat() == InMemoryFormat.NATIVE).count();
+        return mapConfigs.stream().map(MapConfig::getInMemoryFormat)
+                .filter(inMemoryFormat -> inMemoryFormat == InMemoryFormat.NATIVE).count();
+    }
+
+    static Predicate<MapConfig> isMapStoreEnabled() {
+        return p -> p.getMapStoreConfig().isEnabled();
+    }
+
+    static Predicate<MapConfig> isMapStoreDisabled() {
+        return p -> !p.getMapStoreConfig().isEnabled();
+    }
+
+    static class LatencyInfo {
+
+        private long totalLatency;
+        private long operationCount;
+
+        LatencyInfo(long totalLatency, long operationCount) {
+            this.totalLatency = totalLatency;
+            this.operationCount = operationCount;
+        }
+
+        void setLatency(long totalLatency, long totalOperation) {
+            this.totalLatency = totalLatency;
+            this.operationCount = totalOperation;
+        }
+
+        void add(long totalLatency, long operationCount) {
+            setLatency(this.totalLatency + totalLatency,
+                    this.operationCount + operationCount);
+        }
+
+        Long calculateAverage() {
+            return this.operationCount == 0 ? -1 : (this.totalLatency / this.operationCount);
+        }
+    }
+
+
+    private long mapPutLatency(Node node, Predicate<MapConfig> predicate) {
+
+        LatencyInfo putLatencyInfo = new LatencyInfo(0L, 0L);
+        mapConfigs.stream()
+                .filter(predicate)
+                .map(mapConfig -> node.hazelcastInstance.getMap(mapConfig.getName()).getLocalMapStats())
+                .filter(mapStats -> mapStats.getPutOperationCount() != 0L)
+                .forEach(mapStats -> putLatencyInfo.add(mapStats.getTotalPutLatency(), mapStats.getPutOperationCount()));
+
+        return putLatencyInfo.calculateAverage();
+    }
+
+    private long mapGetLatency(Node node, Predicate<MapConfig> predicate) {
+
+        LatencyInfo getLatencyInfo = new LatencyInfo(0L, 0L);
+        mapConfigs.stream()
+                .filter(predicate)
+                .map(mapConfig -> node.hazelcastInstance.getMap(mapConfig.getName()).getLocalMapStats())
+                .filter(mapStats -> mapStats.getGetOperationCount() != 0L)
+                .forEach(mapStats -> getLatencyInfo.add(mapStats.getTotalGetLatency(), mapStats.getGetOperationCount()));
```

Comment:
Can you extract these very similarly looking stream operation chains into a parameterized shared method? What parameters would you need to do that? You will need a functional approach to solve this - please try to figure it out on your own.